### PR TITLE
Add ResourceName methods to *HttpRequest classes

### PR DESCRIPTION
**Pull request by andreamlin**
Request objects, e.g. CreateTopicRequest, now define methods like getTopicAsTopicName so that generated clients like TopicAdminClient can use ResourceNames.

All DiscoGapic request and message classes have no compile errors.

### CODE REVIEW

**Comment by andreamlin** [comment]
PTAL.

File: src/main/java/com/google/api/codegen/config/DiscoveryField.java [hunk 1]
```diff
@@ -72,6 +73,13 @@ public String getFullName() {
   }
 
   @Override
+  public TypeName getTypeName(ImportTypeTable typeTable) {
+    return typeTable
+        .getTypeTable()
+        .getTypeName(((SchemaTypeTable) typeTable).getFullNameFor(schema));
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/config/DiscoveryField.java]: This has an odd look - I see "type table" referenced three times in this expression..

File: src/main/java/com/google/api/codegen/discogapic/transformer/java/JavaDiscoGapicRequestToViewTransformer.java [hunk 2]
```diff
@@ -107,21 +117,46 @@ public JavaDiscoGapicRequestToViewTransformer(
...
-            JavaFeatureConfig.newBuilder().enableStringFormatFunctions(false).build());
-
-    for (Method method : context.getDocument().methods()) {
-      SchemaTransformationContext requestContext =
-          SchemaTransformationContext.create(method.id(), context.getSchemaTypeTable(), context);
-      StaticLangApiMessageView requestView = generateRequestClass(requestContext, method);
-      surfaceRequests.add(generateRequestFile(requestContext, requestView));
+    DiscoApiModel model = new DiscoApiModel(document);
+
+    for (InterfaceModel apiInterface : model.getInterfaces(productConfig)) {
```
Comment (garrettjonesgoogle) [inline src/main/java/com/google/api/codegen/discogapic/transformer/java/JavaDiscoGapicRequestToViewTransformer.java]: I think this function could benefit from being split up more

Comment (andreamlin) [inline src/main/java/com/google/api/codegen/discogapic/transformer/java/JavaDiscoGapicRequestToViewTransformer.java]: stuffed some implementation details in a private method.

File: src/main/java/com/google/api/codegen/config/DiscoveryField.java (hunk 1 above)
Comment (andreamlin) [inline src/main/java/com/google/api/codegen/config/DiscoveryField.java]: less tables now

**Comment by andreamlin** [comment]
made some code a bit more readable
PTAL.

**Pull request merged**